src/lib/bootloader/device_map.rb

```diff
@@ -92,9 +94,7 @@ def reduce_to_bios_limit
         return false
       end
 
-      grub_devices = @model.grub_devices
-      grub_devices.select! { |d| d.start_with?("hd") }
-      grub_devices.sort_by(&:to_i)
```

Comment:
yes, @teclator already discover it. And test suite do not catch it as it checks only number if it is properly reduced to 8 disks and not if correct ones are removed. ( I know it is discutable, but for me having 8 with wrong order is not so bad as having more then 8 causing grub2 to refuse work ).